src/coldfront_plugin_api/scim_v2/auth_middleware.py

```diff
@@ -0,0 +1,28 @@
+import logging
+
+from django_scim.middleware import SCIMAuthCheckMiddleware
+from mozilla_django_oidc.contrib.drf import OIDCAuthentication
+
+logger = logging.getLogger(__name__)
+
+
+class SCIMColdfrontAuthCheckMiddleware(SCIMAuthCheckMiddleware):
+    def __call__(self, request, *args, **kwargs):
+        response = None
+        if hasattr(self, "process_request"):
+            response = self.process_request(request)
+        if not response:
+            response = self.get_response(request, *args, **kwargs)
+        if hasattr(self, "process_response"):
+            response = self.process_response(request, response)
+        return response
+
+    def process_request(self, request):
+        oidc_auth_obj = OIDCAuthentication()
+        user = oidc_auth_obj.authenticate(request)
+
+        if not user or not user[0].is_staff:
```

Comment:
I think it's better if you assign `request.user` and call the superclasses `process_request` anyway. It will then call `get_is_authenticated_predicate()`. 

This way the only change we are introducing is correctly assigning `request.user` on authentication.

Reply:
I'll go one step beyond that.

You should only do `oidc_auth_obj.authenticate(request)` if there is not a `request.user` at all and only if the OIDC backend is configured.

The first allows it to not run authentication when not necessary (session auth, etc), and the second allows it to work when OIDC isn't configured.

See my pseudocode here https://github.com/nerc-project/coldfront-plugin-api/issues/42#issuecomment-2289663318

Reply:
The second reason is why your unit tests failed.

Reply:
I've made a change so that this middleware will perform OIDC authentication if the `PLUGIN_AUTH_OIDC` env var for Coldfront is set, since this env var implies that the DRC ODIC backend has been added to settings.